share/config.fish

```diff
@@ -195,6 +195,68 @@ function . --description 'Evaluate contents of file (deprecated, see "source")'
 	end
 end
 
+# Try to set the locale from the system configuration if we did not inherit any.
```

Comment:
This is a pretty big block of code dedicated to a single purpose. Seems like a good candidate to put in a `__fish_&ast;` function.